src/main/java/io/cdap/plugin/servicenow/connector/ServiceNowRecordConverter.java

```diff
@@ -34,9 +35,39 @@
  * Utility class for converting the record from ServiceNow data type to CDAP schema data types
  */
 public class ServiceNowRecordConverter {
-  private static final String DATE_PATTERN = "yyyy-MM-dd";
-  private static final String DATE_TIME_PATTERN = "yyyy-MM-dd HH:mm:ss";
-  private static final String TIME_PATTERN = "HH:mm:ss";
+
+  // supported date and time formats.
+  // ref: https://www.servicenow.com/docs/bundle/yokohama-api-reference/page/app-store/
+  // dev_portal/API_reference/GlideDateTime/concept/c_GlideDateTimeAPI.html
+  private static final DateTimeFormatter DATE_TIME_FORMATTER = new DateTimeFormatterBuilder()
+      // Full datetime with seconds
+      .appendOptional(DateTimeFormatter.ofPattern("dd.MM.yyyy hh.mm.ss a"))
+      .appendOptional(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss"))
+      .appendOptional(DateTimeFormatter.ofPattern("dd.MM.yyyy HH:mm:ss"))
+      .appendOptional(DateTimeFormatter.ofPattern("dd.MM.yyyy HH.mm.ss"))
+      .appendOptional(DateTimeFormatter.ofPattern("dd-MM-yyyy HH:mm:ss"))
+      .appendOptional(DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm:ss"))
+      .appendOptional(DateTimeFormatter.ofPattern("MM/dd/yyyy HH:mm:ss"))
+      .appendOptional(DateTimeFormatter.ofPattern("MM-dd-yyyy HH:mm:ss"))
+      // Formats without seconds
+      .appendOptional(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm"))
+      .appendOptional(DateTimeFormatter.ofPattern("MM-dd-yyyy HH:mm"))
+      .appendOptional(DateTimeFormatter.ofPattern("dd-MM-yyyy HH.mm"))
+      .toFormatter();
+
+  private static final DateTimeFormatter DATE_FORMATTER = new DateTimeFormatterBuilder()
+      .appendOptional(DateTimeFormatter.ofPattern("yyyy-MM-dd"))
+      .appendOptional(DateTimeFormatter.ofPattern("dd.MM.yyyy"))
+      .appendOptional(DateTimeFormatter.ofPattern("dd-MM-yyyy"))
+      .appendOptional(DateTimeFormatter.ofPattern("dd/MM/yyyy"))
+      .appendOptional(DateTimeFormatter.ofPattern("MM/dd/yyyy"))
+      .appendOptional(DateTimeFormatter.ofPattern("MM-dd-yyyy"))
+      .toFormatter();
+
+  private static final DateTimeFormatter TIME_FORMATTER = new DateTimeFormatterBuilder()
+      .appendOptional(DateTimeFormatter.ofPattern("HH:mm:ss"))
+      .appendOptional(DateTimeFormatter.ofPattern("HH:mm"))
```

Comment:
unable to find this in reference link pasted above, can you please point me to the doc where these are mentioned?

Reply:
why not `hh:mm:ss a` && `hh.mm.ss a`?

Reply:
Also are all of these tested? We should not do guess work in prod plugins.

Reply:
unit testing is one thing but how do we know if they are actually supported by service now if we have added them based on assumption from datetime patterns. Did we do an e2e test with the actual service now data?